dhall/src/Dhall/TypeCheck.hs

```diff
@@ -2813,17 +2876,21 @@ prettyTypeMessage (MustCombineARecord c expr0 expr1) = ErrorMessages {..}
         \                                 Invalid: This is a union type and not a record \n\
         \                                                                                \n\
         \                                                                                \n\
-        \You tried to combine the following value:                                       \n\
+        \────────────────────────────────────────────────────────────────────────────────\n\
         \                                                                                \n\
-        \" <> txt0 <> "\n\
+        \You tried to use this expression:                                               \n\
+        \                                                                                \n\
+        \" <> insert secondaryExpression <> "\n\
+        \                                                                                \n\
+        \... to update this expression:                                                  \n\
+        \                                                                                \n\
+        \" <> insert primaryExpression <> "\n\
```

Comment:
For `{=} // 1`, the explanation seems the wrong way round:

```
You tried to use this expression:

↳ {=}

... to update this expression:

↳ 1

... which is not a record, but is actually a:

↳ Natural
```

We actually try to use `1` to update  `{=}`.

Not sure how to best resolve that…